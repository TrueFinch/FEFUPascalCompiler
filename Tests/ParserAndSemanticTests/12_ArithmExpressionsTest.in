program a;
var
    I: float;
    D: integer;
    C: integer;
begin
    i := 1.1;
    d := 2;
    c := 3;
    I := (5 * D + C - (C shl 15) / ( (+8 shr 1) * (I + 2) * ((D - -10) div 10))) + C mod 10000;
    Write(I);
end.