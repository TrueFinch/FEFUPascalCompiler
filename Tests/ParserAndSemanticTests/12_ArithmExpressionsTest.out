                                                                         a                                                                          
                                                                         ┼                                                                          
                                                                     MainBlock                                                                      
                      ┌──────────────────────────────────────────────────┴─────────────────────┐                                                    
                     var                                                               CompoundStatement                                            
       ┌──────────────┴──────────────┐          ┌──────┬─────┬─────────────────────────────────┴──────┬────────────────────────────────────────┐    
 SimpleVarDecl  SimpleVarDecl  SimpleVarDecl    :=     :=    :=                                      :=                                      write  
 ┌────┬┴        ┌─────┼        ┌─────┼        ┌─┴─┐   ┌┴─┐  ┌┴─┐  ┌───────────────────────────────────┴┐                                     ┌─┴    
 i  float       d  integer     c  integer     i  1.1  d  2  c  3  i                                    +                                     i      
                                                                                                ┌──────┴────────────────────────────┐               
                                                                                                -                             Cast to float         
                                                                           ┌────────────────────┴──────┐                         ┌──┴               
                                                                     Cast to float                     /                        mod                 
                                                                        ┌──┴          ┌────────────────┴───┐                  ┌──┴─┐                
                                                                        +            shl                   *                  c  10000              
                                                                      ┌─┴──┐        ┌─┴┐             ┌─────┴───────────┐                            
                                                                      *    c        c  15            *                div                           
                                                                     ┌┴─┐                   ┌────────┴──┐            ┌─┴──┐                         
                                                                     5  d                  shr          +            -    10                        
                                                                                           ┌┴─┐  ┌──────┴─┐        ┌─┴┐                             
                                                                                           +  1  i  Cast to float  d  -                             
                                                                                           ┼        ┌─────┴           ┼                             
                                                                                           8        2                 10                            
